xdsl/frontend/code_generation.py

```diff
@@ -49,9 +49,18 @@ class CodegGenerationVisitor(ast.NodeVisitor):
     because inner functions and global variables are not allowed (yet).
     """
 
-    file: str = field(default=None)
+    _file: str | None = field(default=None)
```

Comment:
Is making this optional the only solution?
I think this is supposed to never be `None`, since it is assigned in the `__init__`, right?
Could we just remove the default value instead?

Reply:
It's not assigned in the init, and I'm not sure exactly how it's assigned. I wanted to keep as much of the functionality the same as possible. This is potentially worth an issue to follow up, and make it non-optional

Reply:
Fixed